state/validation/validate.js

```diff
@@ -0,0 +1,157 @@
+const {check, validationResult} = require('express-validator');
+
+exports.createActivitiesValidation = [
+    check("name").notEmpty().withMessage("Nama tidak boleh kosong"),
+    check("zoomLink").notEmpty().withMessage("Link zoom tidak boleh kosong"),
+    check("day").notEmpty().withMessage("Day tidak boleh kosong"),
+    check("quota").notEmpty().withMessage("Jumlah quota tidak boleh kosong"),
+    check("attendanceCode").notEmpty().withMessage("code kehadiran tidak boleh kosong") 
+];
+
+exports.logoValidation = (req, res, next)=>{
+    const logoErrors = [];
+    if(!req.files){
+        logoErrors.push({
+            key: "stateLogo",
+            message: "Gambar Logo tidak boleh kosong"
+        })
+    }
+    else{
+        if(req.files.stateLogo.mimetype !== 'image/png'){
+            if(req.files.stateLogo.mimetype !== 'image/jpg'){
+                if(req.files.stateLogo.mimetype !== 'image/jpeg'){
+                    logoErrors.push({
+                        key: "stateLogo",
+                        message: "Harap menggunakan tipe file png, jpg, atau jpeg"
+                    })
+                }
+            }
+        }
```

Comment:
sangat tidak efektif, karena dia itu jadinya bakal ngecek 3x jika nama file jpeg. (jadi kompleksitasnya `n^3`).

```
for (int i = 0; i < n; i++) {
    for (int j = 0; j < n; j++) {
        for (int k = 0; k < n; k++) {
      }
   }
}

```
const yangDibolehin = ['image/png', 'image/jpg', 'image/jpeg']
check = 0
for (int i = 0; i < len(yangDibolehin); i++) if req.files == yangDibolehin[i] check = 1 break; //aman
if check === 0 returnJenisFileNgaco();
```

Reply:
Kalau kalian tertarik untuk belajar lebih dalem, bisa referensi ke sini: [https://www.youtube.com/watch?v=9TlHvipP5yA&list=PLDN4rrl48XKpZkf03iYFl-O29szjTrs_O&index=6](https://www.youtube.com/watch?v=9TlHvipP5yA&list=PLDN4rrl48XKpZkf03iYFl-O29szjTrs_O&index=6)